data/mock_data.rb

```diff
@@ -1,99 +1,123 @@
 require 'Date'
 
 class MockData
-  def self.get_a_random_date(random = true)
-    if random
-      date_s = "20#{rand(10..21)}-#{rand(1..12)}-#{rand(1..28)}"
-      return Date.strptime(date_s, '%Y-%m-%d')
-    else
-      return Date.strptime('2020-01-01', '%Y-%m-%d')
+
+  def self.mock_generator(eg, mock_name, data_hashes)
+    data_hashes.each_with_object([]) do |hash, mocks|
+      mocks << eg.instance_double(mock_name, hash)
     end
   end
 
-  def self.get_a_random_price
-    (rand(1..120) + (rand(100) / 100.0))
+  def self.invoices_as_mocks(eg, invoice_hashes = invoices_as_hashes)
+    mock_generator(eg, 'Invoice', invoice_hashes)
+  end
+
+  def self.merchants_as_mocks(eg, merchant_hashes = merchants_as_hashes)
+    mock_generator(eg, 'Merchant', merchant_hashes)
+  end
+
+  def self.items_as_mocks(eg, item_hashes = items_as_hashes)
+    mock_generator(eg, 'Item', item_hashes)
+  end
+
+  def self.invoice_items_as_mocks(eg, invoice_item_hashes = invoice_items_as_hashes)
+    mock_generator(eg, 'InvoiceItem', invoice_item_hashes)
   end
 
-  def self.merchants_as_mocks(merchant_hashes)
-    mocked_merchants = []
+  def self.invoice_items_as_hashes(number_of_hashes: 10, random_dates: true,
+                            item_id_range: (1..10), invoice_id_range: (1..10),
+                            quantity: get_a_random_quantity,
+                            unit_price: get_a_random_price)
```

Comment:
Does this need to be formatted or should we just leave it alone, rubocop doesn't seem to be formatting it automatically? Thoughts?